### PR TITLE
Fix some code quality and bug-risk issues

### PR DESCRIPTION
**Pull request by pnijhara**
## Change Summary
- Use literal syntax instead of function calls to create the data structure
- Remove unnecessary comprehension
- Remove assert statement from non-test files

Find the other issues found here - [https://deepsource.io/gh/pnijhara/pydantic/issues/?category=recommended](https://deepsource.io/gh/pnijhara/pydantic/issues/?category=recommended)

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

---

This PR also adds `.deepsource.toml` configuration file to run DeepSource analysis on the repo with. Upon enabling DeepSource, the analysis will run on every PR and commit to detect 560+ types of issues in the changes — including bug risks, anti-patterns, security vulnerabilities, etc.

To enable DeepSource analysis after merging this PR, please follow these steps:

1. [Signup](https://deepsource.io/signup/) on DeepSource with your GitHub account and grant access to this repo.
2. Activate analysis on this repo [here](http://deepsource.io/gh/samuelcolvin/pydantic).

You can also look at the [docs](https://deepsource.io/docs/guides/quickstart.html) for more details. Do let me know if I can be of any help!

### CODE REVIEW

File: pydantic/decorator.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def __init__(self, function: 'Callable'):
                 fields[name] = Tuple[annotation, ...], None
                 takes_args = True
             else:
-                assert p.kind == Parameter.VAR_KEYWORD, p.kind
+                if p.kind != Parameter.VAR_KEYWORD:
+                    raise AssertionError(p.kind)
```
Comment (samuelcolvin) [inline pydantic/decorator.py:83]: This should be a `ValueError` or `TypeError` with a proper error message.

File: .deepsource.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+version = 1
```
Comment (samuelcolvin) [inline .deepsource.toml:1]: please remove this.

If you think pydantic should run deepsource as part of CI:
* please create an issue to discuss it and and explain why it's a good idea (I don't personally think any of the things it has found here are particularly impressive)
* if agreed, we would then need to add setup deepsource to work with github statuses etc.

File: pydantic/networks.py [hunk 3]
```diff
@@ -179,7 +179,8 @@ def validate(cls, value: Any, field: 'ModelField', config: 'BaseConfig') -> 'Any
 
         m = url_regex().match(url)
         # the regex should always match, if it doesn't please report with details of the URL tried
-        assert m, 'URL regex failed unexpectedly'
+        if not m:
+            raise AssertionError('URL regex failed unexpectedly')
```
Comment (samuelcolvin) [inline pydantic/networks.py:183]: again, this should be a sensible exception, perhaps `ValueError`.

**Comment by pnijhara** [comment]
> please fill in the PR template and add a change description.
> 
> Is this a bot? If not, why did you think you didn't need to follow the template provided for pull requests?

Apologies for that!
Doing it right away.

File: .deepsource.toml (hunk 2 above)
Comment (pnijhara) [reply .deepsource.toml:1]: > please remove this.
> 
> If you think pydantic should run deepsource as part of CI:
> 
>     * please create an issue to discuss it and and explain why it's a good idea (I don't personally think any of the things it has found here are particularly impressive)
> 
>     * if agreed, we would then need to add setup deepsource to work with github statuses etc.

Sure! Will open an issue on that.

**Comment by samuelcolvin** [comment]
no updates.

**Comment by pnijhara** [comment]
@samuelcolvin please reopen this PR. Sorry for the inactivity, I was busy with my exams. Can work on this now.

Want to open an issue regarding discussion for integrating deespsource at pydantic.

**Comment by samuelcolvin** [comment]
no reply, I'll re-open once these things are fixed.

**Pull request closed without merging**